mod/display.php

```diff
@@ -80,10 +80,10 @@ function display_init(App $a)
 
 		// Is it an item with uid=0?
 		if (!DBA::isResult($item)) {
-			$item = Item::selectFirstForUser(local_user(), $fields, ['guid' => $a->argv[1], 'private' => [0, 2], 'uid' => 0]);
+			$item = Item::selectFirst($fields, ['guid' => $a->argv[1], 'private' => [0, 2], 'uid' => 0]);
```

Comment:
Why this change? The "selectFirstForUser" respects possibly hidden public content for the given user.

Reply:
Isn't the `uid` parameter (set in the condition) immediately overwritten in `selectFirstForUser`? This is one of the most contentious changes I wanted to talk to you about.

Reply:
We always should use ```selectFirstForUser``` or ```selectForUser``` when we are fetching data for display reasons for a specific user. The difference between the calls with and without ```ForUser``` is that we additionally are querying the 'user-item' table where we store things like hiding posts for specific users.

Reply:
In this case, look at the comment `// Is it an item with uid=0?`, is it compatible with a `selectFirstForUser` call?

Reply:
Absolutely. The item user has nothing to do with the viewers user id. In fact it is especially meant for posts with uid=0.

Reply:
I'm waiting with accepting this PR until this line had been changed back.

Reply:
Reverted.